### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by yujunz** [juicedata/juicesync#12]
```
juicesync s3://*:*@https://s3-us-west-1.amazonaws.com/bucket/prefix/ s3://*:*@https://s3.cn-north-1.amazonaws.com.cn/bucket/prefix/

panic: runtime error: index out of range

goroutine 1 [running]:
github.com/juicedata/juicesync/object.newS3(0xc000023fd0, 0xe, 0x7ffd71fb45de, 0x14, 0x7ffd71fb45f3, 0x28, 0xc0001e9dc8, 0x448247)
	/home/ubuntu/go/src/github.com/juicedata/juicesync/object/s3.go:211 +0x4a6
github.com/juicedata/juicesync/object.CreateStorage(0x7ffd71fb45d9, 0x2, 0xc000023fd0, 0xe, 0x7ffd71fb45de, 0x14, 0x7ffd71fb45f3, 0x28, 0x0, 0xc0001e9e80)
	/home/ubuntu/go/src/github.com/juicedata/juicesync/object/object_storage.go:101 +0xbb
main.createStorage(0x7ffd71fb45d9, 0x93, 0xc000023fc0, 0xe)
	/home/ubuntu/go/src/github.com/juicedata/juicesync/main.go:71 +0x197
main.main()
	/home/ubuntu/go/src/github.com/juicedata/juicesync/main.go:97 +0x110

**Comment by davies** [comment]
s3://*:*@https://s3-us-west-1.amazonaws.com/bucket/prefix/  should be  s3://bucket.s3-us-west-1.amazonaws.com/prefix

**Comment by yujunz** [comment]
The same error

```
juicesync s3://AK:SK@bucket.s3-us-west-1.amazonaws.com s3://AK:SK@bucket.s3.cn-north-1.amazonaws.com.cn
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/juicedata/juicesync/object.newS3(0xc00016e780, 0x23, 0x0, 0x0, 0x0, 0x0, 0xc000279dc8, 0x448247)
	/home/admin/go/src/github.com/juicedata/juicesync/object/s3.go:211 +0x4a6
github.com/juicedata/juicesync/object.CreateStorage(0x7ffd1dd93527, 0x2, 0xc00016e780, 0x23, 0x0, 0x0, 0x0, 0x0, 0xe645a0, 0xc0001e1780)
	/home/admin/go/src/github.com/juicedata/juicesync/object/object_storage.go:101 +0xbb
main.createStorage(0x7ffd1dd93527, 0x8a, 0xe645a0, 0xc0001e1780)
	/home/admin/go/src/github.com/juicedata/juicesync/main.go:71 +0x197
main.main()
	/home/admin/go/src/github.com/juicedata/juicesync/main.go:98 +0x13f
goroutine 1 [running]:
github.com/juicedata/juicesync/object.newS3(0xc00016e780, 0x23, 0x0, 0x0, 0x0, 0x0, 0xc000279dc8, 0x448247)
	/home/admin/go/src/github.com/juicedata/juicesync/object/s3.go:211 +0x4a6
github.com/juicedata/juicesync/object.CreateStorage(0x7ffd1dd93527, 0x2, 0xc00016e780, 0x23, 0x0, 0x0, 0x0, 0x0, 0xe645a0, 0xc0001e1780)
	/home/admin/go/src/github.com/juicedata/juicesync/object/object_storage.go:101 +0xbb
main.createStorage(0x7ffd1dd93527, 0x8a, 0xe645a0, 0xc0001e1780)
	/home/admin/go/src/github.com/juicedata/juicesync/main.go:71 +0x197
main.main()
	/home/admin/go/src/github.com/juicedata/juicesync/main.go:98 +0x13
```

**Comment by davies** [comment]
Fixed by caaa5fdd1f0215115e7922ea1fc15b4e597f382f
